docker/configure_workers_and_start.py

```diff
@@ -314,6 +363,20 @@ def add_sharding_to_shared_config(
             "port": worker_port,
         }
 
+    elif worker_type in ["account_data", "presence", "receipts", "to_device", "typing"]:
```

Comment:
Is there a chance to deduplicate code to:
https://github.com/matrix-org/synapse/blob/92a1345f27ddc8d3477980cb2c364f0c7e3552ad/docker/configure_workers_and_start.py#L353-L365

Reply:
Unfortunately, no. All stream writers except for event persister have client endpoints to handle. I didn't want to include weird extra conditionals for that. I'll take suggestions though, if I can follow it.

Reply:
No, I'm sorry. That's wrong. It's because event_persister isn't the same name as the stream type. Event persisters live under events

Reply:
IMO you can change this line:
https://github.com/matrix-org/synapse/blob/92a1345f27ddc8d3477980cb2c364f0c7e3552ad/docker/configure_workers_and_start.py#L178
Rename `event_persister` to `events`.

https://github.com/matrix-org/synapse/blob/develop/docs/workers.md#stream-writers
`event_persister` is only an example for the `worker_name`.

The counting and adding numbers to the `worker_name` happens there:
https://github.com/matrix-org/synapse/blob/92a1345f27ddc8d3477980cb2c364f0c7e3552ad/docker/configure_workers_and_start.py#L495-L503

Reply:
I considered that, but I didn't want to create confusion with `event_creator`. I thought that distinction was made for a reason in the past?

Reply:
This is a question for @anoadragon453
- #9162

Reply:
Excellent information. I wish descriptions like these were in the documentation. I did read somewhere that both `event_`* were `generic_workers` now, but I do not recall where that was. I believe at this point, the only workers that aren't generic is `media_repository`, `federation_sender` and `pusher`.

(Also thanks for justifying my thoughts to *not* change it)

Reply:
Might refactor this in a separate PR. I think there might be better ways to optimize, but if I do it I'll have to put more thought into how it should look.